internal/controller/apiextensions/claim/reconciler.go

```diff
@@ -104,18 +103,32 @@ func (fn CompositeSyncerFn) Sync(ctx context.Context, cm *claim.Unstructured, xr
 	return fn(ctx, cm, xr)
 }
 
+// A ConnectionSecretOwner may create and manage a connection secret in any
+// namespace.
+type ConnectionSecretOwner interface {
+	resource.Object
+	resource.ConnectionSecretWriterTo
+}
+
+// A LocalConnectionSecretOwner may create and manage a connection secret in its
+// own namespace.
+type LocalConnectionSecretOwner interface {
+	resource.Object
+	resource.LocalConnectionSecretWriterTo
+}
```

Comment:
We could go back to using the ones defined in runtime's `pkg/resource` once those types drop ESS support.

That said, I'm a believer in defining the interfaces you use close to where you use them. So maybe we just leave it like this. I don't think it matters that much either way.